nature/views.py

```diff
@@ -6,14 +6,34 @@
 from django.views.generic import DetailView
 from django.views import View
 
-from .models import Feature
+from .models import Feature, Species
 
 
 class FeatureReportView(DetailView):
     queryset = Feature.objects.open_data()
     template_name = 'nature/feature-report.html'
 
 
+class SpeciesReportView(DetailView):
+    queryset = Species.objects.open_data()
+    template_name = 'nature/species-report.html'
+
+    def get_context_data(self, **kwargs):
+        context = {}
+        if self.object:
+            context['feature_classes'] = {}
+            for observation in self.object.observations.all():
+                feature_class = observation.feature.feature_class
+                if feature_class.id not in context['feature_classes'].keys():
+                    context['feature_classes'][feature_class.id] = {
+                        'name': feature_class.name,
+                        'observations': [],
+                    }
+                context['feature_classes'][feature_class.id]['observations'].append(observation)
+        context.update(kwargs)
+        return super().get_context_data(**context)
```

Comment:
I think it's generally a good idea to call `get_context_data` base implementation first and then update the context in child class. This way the context variable overriding becomes explicit.

Reply:
I'm following the structure that Django is using the for the function here. They are calling the super call last here as well. That is why I'm doing this.

Django implementation:
```
    def get_context_data(self, **kwargs):
        """
        Insert the single object into the context dict.
        """
        context = {}
        if self.object:
            context['object'] = self.object
            context_object_name = self.get_context_object_name(self.object)
            if context_object_name:
                context[context_object_name] = self.object
        context.update(kwargs)
        return super(SingleObjectMixin, self).get_context_data(**context)
```

Reply:
But I can flip it around and do it the other way.

Reply:
The base class of SingleObjectMixin will check if the view context variable is already set and only set it when it's not available, so I think it's fine there. But other views bases generally do not do this.